### DISCUSSION TITLE
MQTT

### DISCUSSION THREAD

**Discussion by G4KCM** [theyosh/TerrariumPI#702] [Q&A] [answered]
Hi,

Firstly thank you for the great work with TerrariumPi, I have for many years been using Node-Red on my home server remotely controlling my wifes Crested Gecko terrarium and was about to embark on a project to migrate it to one local controller and then I find TerrariumPi which is just perfect. So most sincerely thank you for a great project.

I have set it all up on test and will shortly install the system.  I do though have a question regarding MQTT.  I have a central MQTT server which I use as common hub for all the different devices around the house and would like to broadcast the sensor values and relay states from TerrariumP but I am uncertain what I have to do in order to broacast the values every thirty seconds.  I looked at an earlier thread discussing MQTT but it didn't really help. 

So, can you please tell me what I have to set in order to broadcast sensor and relay values once every thirty seconds?

Many Thanks - Clive

**Comment by martynwheeler** [comment]
Hey Clive, I do something similar. I have a python script running on the
terrarium pi every minute that grabs data from the API

https://theyosh.github.io/TerrariumPI/api/

The script then pushes the data using mqtt to a central server.

I can share my script if that helps

Martyn

**Comment by theyosh** [comment]
The idea is that you setup this in the notification area. There you can specify which services you want to use (MQTT) and what info you want to be sent to it. You can send a message every 30 seconds when a sensor is updated for example.

**Reply by G4KCM** [reply]
Hi theyosh thanks for the response I tried a 30 sec sensor update message containing-  gecko/sensor/${name} ${value} which I thought might send all the sensors but nothing happened.  Can you give me an example of what I should put there?

**Comment by martynwheeler** [comment]
That sounds even easier

**Comment by martynwheeler** [comment]
It is not clear how to set it up though.  All my mqtt messages are of 
the form home/area/sensor = {data: value}.  How is this done in 
terrarium pi?

**Comment by martynwheeler** [comment]
> 

Here is my python script which runs as a cron job every minute.  It is probably overkill and the notifications built in are probably better.
```python
#!/usr/bin/python3

# see https://github.com/alinleonard/LCD-I2C-Raspberry-Pi-2/blob/master/LCDI2C_backpack.py

# Import Libraries
import requests
import time
import paho.mqtt.client as mqtt
import json

def publishData(data):
    client = mqtt.Client()
    client.connect("192.168.86.22", 1883, 60)
    client.publish("home/front_room/ds18b20/heater", json.dumps(data));
    client.disconnect();


def main():
    r = requests.get('http://localhost:8090/api/sensors/')
    sensors = r.json()['data']
    data = {}
    for sensor in sensors:
        if sensor['name'] == 'Main Temperature':
            data['temperature'] = round(sensor['value'], 1)
        if sensor['name'] == 'Cool Area':
            pass
        if sensor['name'] == 'Humidity':
            pass
    r = requests.get('http://localhost:8090/api/relays/')
    switches = r.json()['data']
    for switch in switches:
        if switch['name'] == 'MPDMv4.1.3':
            data['heater'] = switch['value']
    publishData(data)

if __name__ == '__main__':
    main()
```

**Reply by G4KCM** [reply]
Thanks Martyn that's great.  I had tried a notification 30 sec update message with "gecko/sensor/${name} ${value}"  in it which I thought might send the values but nothing emerged.  I had thoughts along using the api and telegraf or awk but your script looks lke an easier option if I can't get the message to work. Have asked theyosh if he could give me an example.  

Cheers - Clive

**Comment by martynwheeler** [comment]
I think you may have to restart the terrarium pi service to connect to the MQTT broker.  Mine is now sending out messages.  It seems that the topic is fixed in the code in the form:
terrariumpi/sensor/update/848873668075f854bc07853fb386b4de
and the subject and message are added as fields to the MQTT data.  Would be nice to be able to customise the topic but it looks good.  Just need to grab the packet in node-red now.

**Reply by theyosh** [reply]
>  terrariumpi/sensor/update/848873668075f854bc07853fb386b4de

The last part is the ID of the sensor if you not already guessed..

**Comment by theyosh** [comment, marked as answer]
I guess connecting to a MQTT server should not be a problem. Then I used the sensor update message:

![image](https://user-images.githubusercontent.com/8080177/167948276-88931ed1-bc05-4eea-807c-af9966249ac8.png)

This will produce:

![image](https://user-images.githubusercontent.com/8080177/167948444-817837ca-5c7d-43d8-beff-61520fd6e452.png)

The message will contain all the available data. And the custom message is available in the variable: `message`.

The topics are fixed, as it will become to complex for the UI. And it does not work well with other notification services.

**Reply by G4KCM** [reply]
Great! I’m up and running.  Bit of number bashing in node-red now and I can use the values in my existing influx db.

Thank you

**Reply by martynwheeler** [reply]
> Great! I’m up and running. Bit of number bashing in node-red now and I can use the values in my existing influx db.
> 
> Thank you

I added a function in node-red to change the message topic to my standard format and can now switch off my python script.  Thanks Yosh.

**Reply by sni89** [reply]
> Great! I’m up and running. Bit of number bashing in node-red now and I can use the values in my existing influx db.
> 
> Thank you

Would you mind showing me your node red setup? I am starting with Mqtt and managed to setup a broker and let some devices talk to each other but somehow TP messages won’t arrive 😢

**Reply by G4KCM** [reply]
Hope this helps….

[{"id":"9e06a207ddb25401","type":"tab","label":"Marley's New Cage","disabled":false,"info":""},{"id":"e210bf2943a09b9b","type":"mqtt in","z":"9e06a207ddb25401","name":"Marley's Sensors","topic":"terrariumpi/sensor/update/#","qos":"2","datatype":"auto","broker":"cc754b4b7ab18e48","nl":false,"rap":false,"inputs":0,"x":100,"y":80,"wires":[["739b414340177346"]]},{"id":"739b414340177346","type":"json","z":"9e06a207ddb25401","name":"extract JSON ","property":"payload","action":"obj","pretty":false,"x":150,"y":140,"wires":[["c2f6c9a5ed2c959d"]]},{"id":"c2f6c9a5ed2c959d","type":"function","z":"9e06a207ddb25401","name":"Get Values","func":"// var p1 = Number(msg.payload.value);\nvar p1 = msg.payload.value;\nvar p2 = msg.payload.name;\nmsg.payload = parseFloat(p1);\nmsg.topic = p2;\nreturn msg;\n\n//var T1 = Number(msg.payload[\"DS18B20-1\"].Temperature);\n//var T2 = Number(msg.payload[\"DS18B20-2\"].Temperature);\n// var T3 = Number(msg.payload[\"DS18B20-3\"].Temperature);\n//var T4 = Number(msg.payload[\"DS18B20-4\"].Temperature);\n//var AV = (T1 + T2 + T4)/3;\n//msg1 = {\"topic\":\"setCurrent\", \"payload\":AV};\n//msg2 = {\"topic\":\"Temp-5\", \"payload\":msg.payload[\"DS18B20-5\"].Temperature};\n//msg3 = {\"topic\":\"setCurrent\", \"payload\":msg.payload[\"AM2301\"].Humidity};\n// msg3 = {\"topic\":\"Humidity\", \"payload\":\"0\"};\n//msg4 = {\"topic\":\"Temp-3\", \"payload\":msg.payload[\"DS18B20-3\"].Temperature};\n//flow.set('humidity', String(msg3.payload));\n//return\n// [msg1, msg2, msg3, msg4];\n\n// var numInput = Number(msg.payload);\n// var R0 = 10000;\n// var R = R0 / ((1023 / numInput)-1);\n// var T0 = 273 + 25;\n// var B = 3950;\n// var T =  1 / ( (1/T0) + (1/B) * Math.log(R/R0) );\n// msg.payload = T;\n// return msg;\n\n","outputs":1,"noerr":0,"initialize":"","finalize":"","x":190,"y":200,"wires":[["5da1a10fe6c4929e","5c14738423db473a"]]},{"id":"5da1a10fe6c4929e","type":"debug","z":"9e06a207ddb25401","name":"","active":false,"tosidebar":true,"console":false,"tostatus":false,"complete":"false","statusVal":"","statusType":"auto","x":350,"y":80,"wires":[]},{"id":"5c14738423db473a","type":"switch","z":"9e06a207ddb25401","name":"Topic Filter","property":"topic","propertyType":"msg","rules":[{"t":"cont","v":"Humidity","vt":"str"},{"t":"cont","v":"Temperature Mid Cage","vt":"str"},{"t":"cont","v":"1-Wire sensor Mid","vt":"str"},{"t":"cont","v":"1-Wire sensor Floor","vt":"str"},{"t":"cont","v":"Humidifier Level","vt":"str"}],"checkall":"true","repair":false,"outputs":5,"x":210,"y":280,"wires":[["017909a0e26de330"],["c88f0e3e2e6a715b"],["529172aac1d9e085"],["771877b71979f7bd"],["0729d62152477604"]]},{"id":"9bc8d3cafef57d8d","type":"debug","z":"9e06a207ddb25401","name":"","active":false,"tosidebar":true,"console":false,"tostatus":false,"complete":"false","statusVal":"","statusType":"auto","x":910,"y":160,"wires":[]},{"id":"9b5369e63de51025","type":"debug","z":"9e06a207ddb25401","name":"","active":false,"tosidebar":true,"console":false,"tostatus":false,"complete":"false","statusVal":"","statusType":"auto","x":910,"y":240,"wires":[]},{"id":"a44aaed805a7e964","type":"debug","z":"9e06a207ddb25401","name":"","active":false,"tosidebar":true,"console":false,"tostatus":false,"complete":"false","statusVal":"","statusType":"auto","x":910,"y":360,"wires":[]},{"id":"525bc535b45d89b1","type":"debug","z":"9e06a207ddb25401","name":"","active":false,"tosidebar":true,"console":false,"tostatus":false,"complete":"false","statusVal":"","statusType":"auto","x":910,"y":440,"wires":[]},{"id":"42542578cbf96eb5","type":"throttle","z":"9e06a207ddb25401","name":"","throttleType":"time","timeLimit":"30","timeLimitType":"seconds","countLimit":0,"blockSize":0,"locked":false,"x":670,"y":160,"wires":[["9bc8d3cafef57d8d","ca223502.54a8c8","11a83d63.cb4823"]]},{"id":"52ef10a5b443eb9f","type":"throttle","z":"9e06a207ddb25401","name":"","throttleType":"time","timeLimit":"30","timeLimitType":"seconds","countLimit":0,"blockSize":0,"locked":false,"x":670,"y":240,"wires":[["9b5369e63de51025","1b5038ea.f62a57"]]},{"id":"ac7fdaa765c25496","type":"throttle","z":"9e06a207ddb25401","name":"","throttleType":"time","timeLimit":"30","timeLimitType":"seconds","countLimit":0,"blockSize":0,"locked":false,"x":710,"y":360,"wires":[["a44aaed805a7e964","a17789b2.04bee8"]]},{"id":"9c4ccae1e245eecf","type":"throttle","z":"9e06a207ddb25401","name":"","throttleType":"time","timeLimit":"30","timeLimitType":"seconds","countLimit":0,"blockSize":0,"locked":false,"x":710,"y":400,"wires":[["525bc535b45d89b1","daca839.d237a8"]]},{"id":"d3c4585349c62e8e","type":"mqtt in","z":"9e06a207ddb25401","name":"Marley's Relays","topic":"terrariumpi/relay/update/#","qos":"2","datatype":"auto","broker":"cc754b4b7ab18e48","inputs":0,"x":100,"y":380,"wires":[["cbc3f113850cdbaf"]]},{"id":"cbc3f113850cdbaf","type":"json","z":"9e06a207ddb25401","name":"extract JSON ","property":"payload","action":"obj","pretty":false,"x":150,"y":440,"wires":[["1897e760130ce19a"]]},{"id":"1897e760130ce19a","type":"function","z":"9e06a207ddb25401","name":"Get Values","func":"// var p1 = Number(msg.payload.value);\nvar p1 = msg.payload.value;\nvar p2 = msg.payload.name;\nmsg.payload = parseFloat(p1.toFixed(0));\nmsg.topic = p2;\nreturn msg;\n\n//var T1 = Number(msg.payload[\"DS18B20-1\"].Temperature);\n//var T2 = Number(msg.payload[\"DS18B20-2\"].Temperature);\n// var T3 = Number(msg.payload[\"DS18B20-3\"].Temperature);\n//var T4 = Number(msg.payload[\"DS18B20-4\"].Temperature);\n//var AV = (T1 + T2 + T4)/3;\n//msg1 = {\"topic\":\"setCurrent\", \"payload\":AV};\n//msg2 = {\"topic\":\"Temp-5\", \"payload\":msg.payload[\"DS18B20-5\"].Temperature};\n//msg3 = {\"topic\":\"setCurrent\", \"payload\":msg.payload[\"AM2301\"].Humidity};\n// msg3 = {\"topic\":\"Humidity\", \"payload\":\"0\"};\n//msg4 = {\"topic\":\"Temp-3\", \"payload\":msg.payload[\"DS18B20-3\"].Temperature};\n//flow.set('humidity', String(msg3.payload));\n//return\n// [msg1, msg2, msg3, msg4];\n\n// var numInput = Number(msg.payload);\n// var R0 = 10000;\n// var R = R0 / ((1023 / numInput)-1);\n// var T0 = 273 + 25;\n// var B = 3950;\n// var T =  1 / ( (1/T0) + (1/B) * Math.log(R/R0) );\n// msg.payload = T;\n// return msg;\n\n","outputs":1,"noerr":0,"initialize":"","finalize":"","x":190,"y":500,"wires":[["fdf81b1ace9230b1","46b4916f.29d45"]]},{"id":"fdf81b1ace9230b1","type":"debug","z":"9e06a207ddb25401","name":"","active":false,"tosidebar":true,"console":false,"tostatus":false,"complete":"false","statusVal":"","statusType":"auto","x":170,"y":560,"wires":[]},{"id":"0729d62152477604","type":"range","z":"9e06a207ddb25401","minin":"2.5","maxin":"14","minout":"100","maxout":"0","action":"clamp","round":true,"property":"payload","name":"Humidifier Level %","x":490,"y":440,"wires":[["2427ac40a43e809f","3d0b5a35.19ba56"]]},{"id":"2427ac40a43e809f","type":"debug","z":"9e06a207ddb25401","name":"","active":false,"tosidebar":true,"console":false,"tostatus":false,"complete":"false","statusVal":"","statusType":"auto","x":910,"y":520,"wires":[]},{"id":"017909a0e26de330","type":"average","z":"9e06a207ddb25401","name":"Av Humidity","topic":"Av-Humidity","x":510,"y":160,"wires":[["42542578cbf96eb5"]]},{"id":"c88f0e3e2e6a715b","type":"average","z":"9e06a207ddb25401","name":"Av Mid Temp","topic":"Av-MidTemp","x":510,"y":240,"wires":[["52ef10a5b443eb9f"]]},{"id":"529172aac1d9e085","type":"average","z":"9e06a207ddb25401","name":"Av 1-Wire Mid Temp","topic":"Av-1W-MidTemp","x":500,"y":360,"wires":[["ac7fdaa765c25496","8eeea096.3c932"]]},{"id":"771877b71979f7bd","type":"average","z":"9e06a207ddb25401","name":"Av-1W-Floor Temp","topic":"Av-1W-FloorTemp","x":490,"y":400,"wires":[["9c4ccae1e245eecf"]]},{"id":"a17789b2.04bee8","type":"influxdb out","z":"9e06a207ddb25401","influxdb":"a3f999.c2c87668","name":"Cage Temp PV","measurement":"CageTemp","precision":"","retentionPolicy":"","database":"","retentionPolicyV18Flux":"","org":"","bucket":"","x":920,"y":400,"wires":[]},{"id":"daca839.d237a8","type":"influxdb out","z":"9e06a207ddb25401","influxdb":"a3f999.c2c87668","name":"Floor Temp","measurement":"FloorTemp","precision":"","retentionPolicy":"","database":"","retentionPolicyV18Flux":"","org":"","bucket":"","x":910,"y":480,"wires":[]},{"id":"6719ea6.0970e14","type":"influxdb out","z":"9e06a207ddb25401","influxdb":"a3f999.c2c87668","name":"TP Cage Heat","measurement":"TPHeat","precision":"","retentionPolicy":"","database":"","retentionPolicyV18Flux":"","org":"","bucket":"","x":920,"y":920,"wires":[]},{"id":"aab940b5.29ee7","type":"influxdb out","z":"9e06a207ddb25401","influxdb":"a3f999.c2c87668","name":"TP Cage Floor Heat","measurement":"TPFloorHeat","precision":"","retentionPolicy":"","x":900,"y":960,"wires":[]},{"id":"889c9bda.6613d8","type":"influxdb out","z":"9e06a207ddb25401","influxdb":"a3f999.c2c87668","name":"Basking Light","measurement":"BaskingOnOFF","precision":"","retentionPolicy":"","database":"","retentionPolicyV18Flux":"","org":"","bucket":"","x":920,"y":880,"wires":[]},{"id":"ff663c5c.0dd33","type":"influxdb out","z":"9e06a207ddb25401","influxdb":"a3f999.c2c87668","name":"Main Light ","measurement":"MainLight","precision":"","retentionPolicy":"","database":"","retentionPolicyV18Flux":"","org":"","bucket":"","x":930,"y":840,"wires":[]},{"id":"ca223502.54a8c8","type":"influxdb out","z":"9e06a207ddb25401","influxdb":"a3f999.c2c87668","name":"Cage Humidity","measurement":"CageHumi","precision":"","retentionPolicy":"","database":"","retentionPolicyV18Flux":"","org":"","bucket":"","x":920,"y":200,"wires":[]},{"id":"f63b63c5.85616","type":"influxdb out","z":"9e06a207ddb25401","influxdb":"a3f999.c2c87668","name":"Fan ON/OFF","measurement":"FanOnOFF","precision":"","retentionPolicy":"","database":"","retentionPolicyV18Flux":"","org":"","bucket":"","x":930,"y":1040,"wires":[]},{"id":"f2668264.82b94","type":"influxdb out","z":"9e06a207ddb25401","influxdb":"a3f999.c2c87668","name":"TP Cage Humi","measurement":"TPHumi","precision":"","retentionPolicy":"","database":"","retentionPolicyV18Flux":"","org":"","bucket":"","x":920,"y":1000,"wires":[]},{"id":"46b4916f.29d45","type":"switch","z":"9e06a207ddb25401","name":"Topic Filter","property":"topic","propertyType":"msg","rules":[{"t":"cont","v":"Relay 1","vt":"str"},{"t":"cont","v":"Relay 2","vt":"str"},{"t":"cont","v":"Relay 3","vt":"str"},{"t":"cont","v":"Relay 4","vt":"str"},{"t":"cont","v":"Relay 5","vt":"str"},{"t":"cont","v":"Relay 6","vt":"str"},{"t":"cont","v":"Relay 7","vt":"str"},{"t":"cont","v":"Relay 8","vt":"str"}],"checkall":"true","repair":false,"outputs":8,"x":210,"y":960,"wires":[["ff663c5c.0dd33"],["889c9bda.6613d8"],["85e8146e.ed9098"],["ccd83f4a.8277b"],["b835fc8a.36054"],["f63b63c5.85616"],[],[]]},{"id":"b835fc8a.36054","type":"range","z":"9e06a207ddb25401","minin":"0","maxin":"100","minout":"15","maxout":"20","action":"scale","round":true,"property":"payload","name":"Shift Scale","x":670,"y":1000,"wires":[["f2668264.82b94"]]},{"id":"85e8146e.ed9098","type":"range","z":"9e06a207ddb25401","minin":"0","maxin":"100","minout":"0","maxout":"5","action":"scale","round":true,"property":"payload","name":"Shift Scale","x":670,"y":920,"wires":[["6719ea6.0970e14"]]},{"id":"ccd83f4a.8277b","type":"range","z":"9e06a207ddb25401","minin":"0","maxin":"100","minout":"7.5","maxout":"12.5","action":"scale","round":true,"property":"payload","name":"Shift Scale","x":670,"y":960,"wires":[["aab940b5.29ee7"]]},{"id":"8205f149.61c67","type":"ramp-thermostat","z":"9e06a207ddb25401","name":"Humidity Profile","profile":"4091eb7c.73de34","hysteresisplus":0,"hysteresisminus":0,"x":700,"y":60,"wires":[[],[],["a4004f99.1a991","12207069.399ab"]]},{"id":"a4004f99.1a991","type":"influxdb out","z":"9e06a207ddb25401","influxdb":"a3f999.c2c87668","name":"Cage Humidity SP","measurement":"CageHumiSet","precision":"","retentionPolicy":"","database":"","retentionPolicyV18Flux":"","org":"","bucket":"","x":930,"y":120,"wires":[]},{"id":"12207069.399ab","type":"debug","z":"9e06a207ddb25401","name":"","active":false,"tosidebar":true,"console":false,"tostatus":false,"complete":"false","statusVal":"","statusType":"auto","x":910,"y":80,"wires":[]},{"id":"93022a5a.da7048","type":"ramp-thermostat","z":"9e06a207ddb25401","name":"Temp Profile","profile":"410d9945.6239c8","hysteresisplus":0,"hysteresisminus":0,"x":690,"y":300,"wires":[[],[],["79bfef30.ffed","f7aee6df.cf3808"]]},{"id":"79bfef30.ffed","type":"influxdb out","z":"9e06a207ddb25401","influxdb":"a3f999.c2c87668","name":"Cage Temp SP","measurement":"CageTempSet","precision":"","retentionPolicy":"","database":"","retentionPolicyV18Flux":"","org":"","bucket":"","x":920,"y":320,"wires":[]},{"id":"f7aee6df.cf3808","type":"debug","z":"9e06a207ddb25401","name":"","active":false,"tosidebar":true,"console":false,"tostatus":false,"complete":"false","statusVal":"","statusType":"auto","x":910,"y":280,"wires":[]},{"id":"11a83d63.cb4823","type":"change","z":"9e06a207ddb25401","name":"SetTopic","rules":[{"t":"set","p":"topic","pt":"msg","to":"setCurrent","tot":"str"}],"action":"","property":"","from":"","to":"","reg":false,"x":540,"y":100,"wires":[["8205f149.61c67"]]},{"id":"1b5038ea.f62a57","type":"change","z":"9e06a207ddb25401","name":"SetTopic","rules":[{"t":"set","p":"topic","pt":"msg","to":"setCurrent","tot":"str"}],"action":"","property":"","from":"","to":"","reg":false,"x":530,"y":302,"wires":[["93022a5a.da7048"]]},{"id":"8eeea096.3c932","type":"switch","z":"9e06a207ddb25401","name":"Test Value","property":"payload","propertyType":"msg","rules":[{"t":"gte","v":"28","vt":"num"},{"t":"lte","v":"18","vt":"num"}],"checkall":"true","repair":false,"outputs":2,"x":710,"y":680,"wires":[["8efa5a7a.3f88d8"],["9c1c6d94.9ace7"]]},{"id":"a714708e.4ede4","type":"prowl","z":"9e06a207ddb25401","title":"","priority":"2","url":"","name":"Prowl Message","x":1340,"y":640,"wires":[]},{"id":"d6f3776a.bd00e8","type":"throttle","z":"9e06a207ddb25401","name":"","throttleType":"time","timeLimit":"60","timeLimitType":"minutes","countLimit":0,"blockSize":0,"locked":false,"x":1130,"y":680,"wires":[["a714708e.4ede4"]]},{"id":"8efa5a7a.3f88d8","type":"change","z":"9e06a207ddb25401","name":"High Temp Alert ","rules":[{"t":"set","p":"topic","pt":"msg","to":"Urgent Gecko Alert","tot":"str"},{"t":"set","p":"payload","pt":"msg","to":"The cage temperature is greater than 28°C.  Please cool Marley down!!","tot":"str"}],"action":"","property":"","from":"","to":"","reg":false,"x":940,"y":660,"wires":[["d6f3776a.bd00e8"]]},{"id":"9c1c6d94.9ace7","type":"change","z":"9e06a207ddb25401","name":"Low Temp Alert","rules":[{"t":"set","p":"topic","pt":"msg","to":"Urgent Gecko Alert","tot":"str"},{"t":"set","p":"payload","pt":"msg","to":"The cage temperature is less than 18°C.  Please warm Marley up!!","tot":"str"}],"action":"","property":"","from":"","to":"","reg":false,"x":940,"y":700,"wires":[["d6f3776a.bd00e8"]]},{"id":"e511bf84.b0064","type":"change","z":"9e06a207ddb25401","name":"Low Level Alert 1","rules":[{"t":"set","p":"topic","pt":"msg","to":"Gecko Alert","tot":"str"},{"t":"set","p":"payload","pt":"msg","to":"The humidifer is at less than 20% level. Please refill ASAP..","tot":"str"}],"action":"","property":"","from":"","to":"","reg":false,"x":930,"y":580,"wires":[["8f27e119.87626"]]},{"id":"3d0b5a35.19ba56","type":"switch","z":"9e06a207ddb25401","name":"Test Value","property":"payload","propertyType":"msg","rules":[{"t":"btwn","v":"12","vt":"num","v2":"6","v2t":"num"},{"t":"btwn","v":"5","vt":"num","v2":"1","v2t":"num"}],"checkall":"true","repair":false,"outputs":2,"x":710,"y":600,"wires":[["e511bf84.b0064"],["cc4119d6.b589d8"]]},{"id":"8f27e119.87626","type":"throttle","z":"9e06a207ddb25401","name":"","throttleType":"time","timeLimit":"8","timeLimitType":"hours","countLimit":0,"blockSize":0,"locked":false,"x":1130,"y":600,"wires":[["a714708e.4ede4"]]},{"id":"cc4119d6.b589d8","type":"change","z":"9e06a207ddb25401","name":"Low Level Alert 2","rules":[{"t":"set","p":"topic","pt":"msg","to":"Gecko Alert","tot":"str"},{"t":"set","p":"payload","pt":"msg","to":"The humidifer is at less than 10% level. Please get your arse in gear and refill it!  Love and kisses Snappy McSnapface","tot":"str"}],"action":"","property":"","from":"","to":"","reg":false,"x":930,"y":620,"wires":[["8f27e119.87626"]]},{"id":"cc754b4b7ab18e48","type":"mqtt-broker","name":"Home MQTT","broker":"192.168.1.22","port":"1883","clientid":"","autoConnect":true,"usetls":false,"protocolVersion":"4","keepalive":"60","cleansession":true,"birthTopic":"","birthQos":"0","birthPayload":"","birthMsg":{},"closeTopic":"","closeQos":"0","closePayload":"","closeMsg":{},"willTopic":"","willQos":"0","willPayload":"","willMsg":{},"userProps":"","sessionExpiry":""},{"id":"a3f999.c2c87668","type":"influxdb","hostname":"a0d7b954-influxdb","port":"8086","protocol":"http","database":"node-red","name":"Influx DB ","usetls":false,"tls":"","influxdbVersion":"1.x","url":"","rejectUnauthorized":false},{"id":"4091eb7c.73de34","type":"profile","name":"New Cage Humidity Profile","time1":"00:00","temp1":"75","time2":"09:30","temp2":"75","time3":"10:00","temp3":"60","time4":"15:00","temp4":"60","time5":"20:00","temp5":"60","time6":"21:00","temp6":"60","time7":"21:30","temp7":"75","time8":"22:00","temp8":"75","time9":"23:00","temp9":"75","time10":"23:59","temp10":"75"},{"id":"410d9945.6239c8","type":"profile","name":"New Cage Temp","time1":"00:00","temp1":"21","time2":"01:00","temp2":"21","time3":"09:30","temp3":"21","time4":"10:00","temp4":"25","time5":"14:00","temp5":"25","time6":"17:00","temp6":"25","time7":"21:00","temp7":"25","time8":"21:30","temp8":"21","time9":"23:00","temp9":"21","time10":"23:59","temp10":"21"}]

**Reply by sni89** [reply]
Could you add a screenshot of your node-red interface maybe? Like how things are connected? I set up a simple testsetup with a Mqtt in and a debug that won’t give me a plain text from my TP. But it does show up all messages other devices send to the broker.

**Reply by G4KCM** [reply]
![33E51307-CD9B-4525-8E3A-324CA6D3518A](https://user-images.githubusercontent.com/17541578/210435680-442af043-e456-4532-bf0e-f83bad0a87e0.png)

**Reply by sni89** [reply]
Sorry I meant the node-red interface like 
![90D406FD-EEDD-4D3E-90EF-A22F2DFBA0DD](https://user-images.githubusercontent.com/56272047/210439704-85bfe4af-d8ec-4d3d-96dd-e376d40bc86e.png)

this is where I get everything to work but no messages incoming from TP. Even I did everything as described above.

**Reply by G4KCM** [reply]
Hope this helps,  but if you cut and paste the flow I sent yesterday as an import into a blank node red page you will see how I extract the values.
![4DD412C7-16EA-4EE2-8FA9-5BEE8E5B2D89](https://user-images.githubusercontent.com/17541578/210441729-3ba0db0b-a8fa-48a5-b687-de9e29f7c251.png)

**Reply by sni89** [reply]
Thanks a lot. i had to replace a thousand `"` with` '`, add a` #` in the topic and it finally works... 
:)

**Comment by martynwheeler** [comment]
I also found that I need to restart after setting up the MQTT notification
in order for it to connect.

**Reply by G4KCM** [reply]
Thanks, all up and running appreciate the help from you both!
